sr_port/mdb_condition_handler.c

```diff
@@ -666,7 +666,7 @@ CONDITION_HANDLER(mdb_condition_handler)
 		if (!repeat_error)
 			/* This has already been done if we are re-throwing the error */
 			outofband_clear();
-		if (!trans_action && !ctrlc_on && !(frame_pointer->type & SFT_DM))
+		if (!trans_action && !dm_action && !(frame_pointer->type & SFT_DM))
```

Comment:
To be sure about the changes, and to minimize the risk of a regression, I would like to see a use case that can only be made to work with the fix and never otherwise.